eng/scripts/Process-PackageReadMe.ps1

```diff
@@ -0,0 +1,91 @@
+#!/bin/env pwsh
+#Requires -Version 7
+
+param(
+    [Parameter(Mandatory=$true)]
+    [string] $InputReadMePath,
+    [Parameter(Mandatory=$true)]
+    [string] $OutputDirectory,
+    [Parameter(Mandatory=$true)]
+    [ValidateSet('nuget','npm','vsix')]
+    [string] $PackageType,
+    [hashtable] $InsertPayload = @{}
+)
+
+$readMeText = Get-Content $InputReadMePath         
+$processedReadMe = @()
+$appendLine = $true
+
+# Remove leading comment block if present
+if ($readMeText[0] -eq "<!--") {
+    for ($i = 0; $i -lt $readMeText.Count; $i++) {
+        if ($readMeText[$i] -eq "-->") {
+            $readMeText = $readMeText | Select-Object -Skip ($i + 1)
+            break
+        }
+    }
+}
+
+foreach ($line in $readMeText) {
+    if ([string]::IsNullOrWhiteSpace($line) -and $appendLine) {
+        $processedReadMe += ''
+        continue
+    }
+
+    if ($line -match "(?i)<!--\s*remove-section:\s*start\s+([^>]+?)\s*-->") {
+        $pkgTypeInfo = $matches[1]
```

Comment:
Seems like a common patter is to match the package type and check if it exists. That seems like a potential helper function and/or regex.